app/src/main/java/com/hedvig/app/feature/language/LanguageAndMarketViewModel.kt

```diff
@@ -145,10 +146,22 @@ class LanguageAndMarketViewModel(
 
     fun loadGeo() {
         viewModelScope.launch {
-            val response = marketRepository.geoAsync().await()
-            preselectedMarket.postValue(response.data()?.geo?.countryISOCode)
-            response.data()?.geo?.let { geo ->
-                updateMarket(Market.valueOf(geo.countryISOCode))
+            val response = runCatching {
+                marketRepository.geoAsync().await()
+            }
+            if (response.isFailure) {
+                response.exceptionOrNull()?.let { e(it) }
+                updateMarket(Market.SE)
+            } else {
+                preselectedMarket.postValue(response.getOrNull()?.data()?.geo?.countryISOCode)
+                response.getOrNull()?.data()?.geo?.let { geo ->
+                    val market: Market = try {
```

Comment:
We can just skip pre-selecting a market if we can't coerce the Geo-information to a market 🙂 so just return early at this point really

Reply:
also same if we get an error in the network request 🙂